hls4ml/templates/vivado/build_prj.tcl

```diff
@@ -161,7 +161,7 @@ if {$opt(reset)} {
 } else {
     open_solution "solution1"
 }
-catch {config_array_partition -maximum_size 4096}
+catch {config_array_partition -maximum_size 8192}
```

Comment:
I think this change should be discussed separately from this PR. We had 4k limit forever, increasing it allows the synthesis to run much longer, take more memory but it is not clear if this ultimately works and meets timing. I wonder why Xilinx set it to 1024 by default and then removed it in Vitis. Either way, I'd be more in favor of this being configurable as "you're on your own if you change it"

Reply:
created a new configurable parameter `MaximumSize` that is by default 4096

Reply:
I know I'm nitpicking here, but can we rename this to `ArrayPartitionLimit` or something more descriptive? MaximumSize doesn't provide information as to what it refers to